Documentation/Books/Manual/Programs/Arangorestore/ParallelRestore.md

```diff
@@ -1,2 +1,198 @@
+var fs = require("fs");
+var print = require("internal").print;
+var exit = require("internal").exit;
+var arangorestore = "arangorestore";
+var env = require("internal").env;
+if (env.hasOwnProperty("ARANGORESTORE")) {
+  arangorestore = env["ARANGORESTORE"];
+}
+
+// Check ARGUMENTS: dumpDir coordinator1 coordinator2 ...
+
+if (ARGUMENTS.length < 2) {
+  print("Need at least two arguments DUMPDIR and COORDINATOR_ENDPOINTS!");
+  exit(1);
+}
+
+var dumpDir = ARGUMENTS[0];
+var coordinators = ARGUMENTS[1].split(",");
+var otherArgs = ARGUMENTS.slice(2);
+
+// Quickly check the dump dir:
+var files = fs.list(dumpDir).filter(f => !fs.isDirectory(f));
+var found = files.indexOf("ENCRYPTION");
+if (found === -1) {
+  print("This directory does not have an ENCRYPTION entry.");
+  exit(2);
+}
+// Remove ENCRYPTION entry:
+files = files.slice(0, found).concat(files.slice(found+1));
+
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].slice(-5) !== ".json") {
+    print("This directory has files which do not end in '.json'!");
+    exit(3);
+  }
+}
+files = files.map(function(f) {
+  var fullName = fs.join(dumpDir, f);
+  var collName = "";
+  if (f.slice(-10) === ".data.json") {
+    var pos;
+    if (f.slice(0, 1) === "_") {  // system collection
+      pos = f.slice(1).indexOf("_") + 1;
+      collName = "_" + f.slice(1, pos);
+    } else {
+      pos = f.indexOf("_");
+      collName = f.slice(0, pos);
+    }
+  }
+  return {name: fullName, collName, size: fs.size(fullName)};
+});
+files = files.sort(function(a, b) { return b.size - a.size; });
+var dataFiles = [];
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].name.slice(-10) === ".data.json") {
+    dataFiles.push(i);
+  }
+}
+
+// Produce the scripts, one for each coordinator:
+var scripts = [];
+for (let i = 0; i < coordinators.length; ++i) {
+  scripts.push([]);
+}
+
+var cnum = 0;
+for (let i = 0; i < dataFiles.length; ++i) {
+  var f = files[dataFiles[i]];
+  scripts[cnum].push(`${arangorestore} --collection ${f.collName} --input-directory ${dumpDir} --server.endpoint ${coordinators[cnum]} ` + otherArgs.join(" "));
+  cnum += 1;
+  if (cnum >= coordinators.length) {
+    cnum = 0;
+  }
+}
+
+for (let i = 0; i < coordinators.length; ++i) {
+  let f = "coordinator_" + i + ".sh";
+  print("Writing file", f, "...");
+  fs.writeFileSync(f, scripts[i].join("\n"));
+}
+EOF
+
+${ARANGOSH} --javascript.execute /tmp/parallelRestore$$.js -- "$@"
+rm /tmp/parallelRestore$$.js
+```
+
+To run this script, all coordinator endpoints of the cluster have to be
+provided. The example below is a template for a cluster with 3 coordinators:
+
+```
+./parallelRestore.js <dump-directory>
+  tcp://<ip-of-coordinator1>:<port of coordinator1>,
+  tcp://<ip-of-coordinator2>:<port of coordinator2>,
+  tcp://<ip-of-coordinator3>:<port of coordinator3>
+  --server.username <username>
+  --server.password <password>
+  --server.database <database_name>
+  --create-collection false
+```
```

Comment:
How to use this snippet? Normally, you can not run JavaScript files directly. Is this to be used together with Arangosh? Node.js? In some config file?

Reply:
it is a #!/bin/sh , unsure why it was named parallelRestore.js, probably the correct name should be parallelRestore.sh as it is bin/sh script .. @maxkernbach it will work if we rename it to parallelRestore.sh, right?

Reply:
the js script is actually passed to arangosh with command ${ARANGOSH} --javascript.execute /tmp/parallelRestore$$.js - so as far as I understand the first script is a bash script that creates a .js script.   So I believe the script can be renamed .sh (and it will create anyway a .js script) - we should document that such script will be created under /tmp and as such this /tmp dir should exist and be writable by the user (the default as far as i know, but just in case) - wdyt